test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -809,6 +809,10 @@ public static boolean randomBoolean() {
         return random().nextBoolean();
     }
 
+    public static Boolean randomOptionalBoolean() {
+        return randomBoolean() ? Boolean.TRUE : randomFrom(Boolean.FALSE, null);
```

Comment:
This changes the statistics of the responses somewhat, what about
```suggestion
        return randomFrom(Boolean.TRUE, Boolean.FALSE, null);
```

Reply:
This was on purpose because otherwise, "falsey" return values (`Boolean.FALSE`, `null`) would have a higher probability. As this is a new method, and mostly used for new code, it doesn't really alter existing behavior. But it does assume that `null` is equivalent to `false`, which may not always be true in all cases.

I'll leave this as-is for now but happy to change if you still think it would be better. But I'll not block merging.